library.js

```diff
@@ -1,44 +1,50 @@
 const inputUser = document.querySelector('.input-user');
 const createBookBtn = document.querySelector('.crt_book');
 const readBtn = document.querySelector('.btn-read');
+const form = document.querySelector('#form');
+
 let inputBarOn = false;
 let currBook = 0;
 let myLibrary = [];
 
 // show and hide the form once toggle
-function displayInput(){
-    inputBarOn = true;
-    if(inputBarOn == true){
-        inputUser.classList.toggle('show');
-    }
+function displayInput() {
+  inputBarOn = true;
+  if (inputBarOn == true) {
+    inputUser.classList.toggle('show');
+  }
 }
 
-function book(name, author, pages){
-    this.name = name
-    this.author = author
-    this.pages = pages
-    this.read = false; // initialize the read status to false
+function book(name, author, pages) {
+  this.name = name;
+  this.author = author;
+  this.pages = pages;
+  this.read = false; // initialize the read status to false
 }
 
-book.prototype.toggleRead = function() {
-    this.read = !this.read; // toggle the read status between true and false
-}
+book.prototype.toggleRead = function () {
+  this.read = !this.read; // toggle the read status between true and false
+};
 
 function createBook() {
-    // Prevent the form from submitting
-
+  if (form.checkValidity()) {
     // get the values from the form inputs
```

Comment:
and this if/else where it checks if the form is valid and if not triggers the popups, please do keep in mind that "form.reportValidity()" is not fully supported on internet explorer browser